File: warmup/sherlocksquares.py
from sys  import stdin
from math import sqrt
from math import ceil

#file = stdin
file = open( r".\data\sherlocksquares.txt" ) #stdin
data = list(map(int, file.read().strip().split()))[1:]
file.close()

def numSquares(mini, maxi):
	cnt = 0
	start = ceil(sqrt(mini))
	while start**2 <= maxi:
		cnt   += 1
		start += 1
	return cnt

# For each test case pair...
for i in range(0, len(data), 2):
	print(numSquares(*data[i:i+2]))

File: warmup/sherlocksquares.go
package main

import (
	"fmt"
	"io/ioutil"
	"math"
	"os"
	"strconv"
	"strings"
)

func main() {
	//file := os.Stdin
	file, _ := os.Open(`.\data\sherlocksquares.txt`)
	rawBytes, _ := ioutil.ReadAll(file)

	cases := strings.Fields(string(rawBytes))[1:]

	for i := 0; i <= len(cases)-2; i += 2 {
		a, _ := strconv.ParseFloat(cases[i+0], 64)
		b, _ := strconv.ParseFloat(cases[i+1], 64)
		fmt.Println(numSquares(a, b))
	}
}

func numSquares(min, max float64) (cnt int) {
	start := math.Ceil(math.Sqrt(min))
	for ; start*start <= max; start++ {
		cnt++
	}
	return
}
